core-tests/src/test/java/org/mule/runtime/core/internal/routing/ChoiceRouterTestCase.java

```diff
@@ -144,6 +149,39 @@ public void failingRoute() throws Exception {
     process(choiceRouter, zapEvent());
   }
 
+  @Test
+  public void subscriberContextPropagation() throws MuleException {
+    final ContextPropagationChecker contextPropagationChecker = new ContextPropagationChecker();
+
+    MessageProcessorChain mp = newChain(empty(), contextPropagationChecker);
+    choiceRouter.addRoute(payloadZapExpression(), mp);
+    initialise();
+
+    final CoreEvent result = just(zapEvent())
```

Comment:
duplicated code. Maybe this assertion should be unified. Verify if this can be done in all the tests.